src/GraphQL/ParseGraphQLServer.js

```diff
@@ -44,6 +45,16 @@ class ParseGraphQLServer {
             this.parseServer.config.maxUploadSize || '20mb'
           ),
         },
+        plugins: [
+          useDisableIntrospection({
+            isDisabled: request => {
+              if (!this.parseServer.config.masterKeyOnlyIntrospection) return false;
+              return (
+                request.headers.get('X-Parse-Master-Key') !== this.parseServer.config.masterKey
+              );
+            },
+          }),
+        ],
```

Comment:
Your PR is totally valid, but i think we should aim further and actually just expose a "graphQLPlugin" option on Parse Server to allow developers to provide any custom logic and plugins to the graphql server. It will be a more flexible approach and open many other use cases support

Reply:
I agree, it looks strange to have a graphQL option on the root level of Parse Server, given that GraphQL should be considered more of an "adapter", even though it is currently not yet implemented as such.

Reply:
Sorry have been on parental leave, so I haven't look at this for a while. I will look at it again. Is your comment still valid ?